nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateFlowFile.java

```diff
@@ -183,12 +200,22 @@ protected Collection<ValidationResult> customValidate(ValidationContext validati
         final List<ValidationResult> results = new ArrayList<>(1);
         final boolean isUnique = validationContext.getProperty(UNIQUE_FLOWFILES).asBoolean();
         final boolean isText = validationContext.getProperty(DATA_FORMAT).getValue().equals(DATA_FORMAT_TEXT);
+        final boolean isExternal = validationContext.getProperty(DATA_FORMAT).getValue().equals(DATA_FORMAT_EXTERNAL_FILE);
         final boolean isCustom = validationContext.getProperty(CUSTOM_TEXT).isSet();
+        final boolean isCustomContent = validationContext.getProperty(CUSTOM_CONTENT).isSet();
 
         if (isCustom && (isUnique || !isText)) {
             results.add(new ValidationResult.Builder().subject("Custom Text").valid(false).explanation("If Custom Text is set, then Data Format must be "
                     + "text and Unique FlowFiles must be false.").build());
         }
+        if (isCustomContent && (isUnique || !isExternal)) {
+            results.add(new ValidationResult.Builder().subject("Custom Content").valid(false).explanation("If Custom Content is set, then Data Format must be "
+                    + "External File and Unique FlowFiles must be false.").build());
+        }
+        if (isCustomContent && isCustom) {
```

Comment:
Do we need this validation explicitly? The `dependsOn` conditionals should result in only on of `Custom Text` or `Custom Content` being available to the user I assume?